soh/soh/Enhancements/randomizer/location_access/overworld/market.cpp

```diff
@@ -138,8 +138,14 @@ void RegionTable_Init_Market() {
 
     areaTable[RR_MARKET_MASK_SHOP] = Region("Market Mask Shop", "Market Mask Shop", {}, NO_DAY_NIGHT_CYCLE, {
         //Events
-        EventAccess(&logic->SkullMask,   []{return logic->SkullMask   || (logic->HasItem(RG_ZELDAS_LETTER) && (ctx->GetOption(RSK_COMPLETE_MASK_QUEST) ||  ChildCanAccess(RR_KAKARIKO_VILLAGE)));}), //RANDOTODO Complete mask quest does not need this location, so should be tied to link's pocket
-        EventAccess(&logic->MaskOfTruth, []{return logic->MaskOfTruth || (logic->SkullMask && (ctx->GetOption(RSK_COMPLETE_MASK_QUEST) || (ChildCanAccess(RR_THE_LOST_WOODS) && logic->CanUse(RG_SARIAS_SONG) && RegionTable(RR_THE_GRAVEYARD)->childDay && ChildCanAccess(RR_HYRULE_FIELD) && logic->StoneCount() == 3)));}),
+        //Currently, mask swap in menu doesn't need access to the mask shop
+        //If it is forced on/a setting, a copy of these events should be added to root
+        //it also doesn't need you to open kak gate, but that might be best treated as a bug
```

Comment:
might be simpler to make this an enforced rando qol thing, especially since with Shuffle Climb it's possible to need Zelda's Letter to reach Great Fairy's Fountain at Hyrule Castle, which wouldn't be allowed logically if player lost Zelda's Letter

Reply:
Yes. Wasn't sure if that's sometimes disabled